File: MediaPlayer/MediaPlayer/Util/Category/UIButton+BackGroundColor.m
//
//  UIButton+BackGroundColor.m
//  StudentLive
//
//  Created by hefanghui on 2017/7/28.
//  Copyright © 2017年 hqyxedu. All rights reserved.
//
//  Main function:设置Button不同状态下的背景色
//
//  Other specifications:
//
//  ************************************************************************

#import "UIButton+BackGroundColor.h"

@implementation UIButton (BackGroundColor)

- (void)setBackgroundColor:(UIColor *)backgroundColor forState:(UIControlState)state {
    [self setBackgroundImage:[UIButton p_imageWithColor:backgroundColor] forState:state];
}

+ (UIImage *)p_imageWithColor:(UIColor *)color {
    CGRect rect = CGRectMake(0.0f, 0.0f, 1.0f, 1.0f);
    UIGraphicsBeginImageContext(rect.size);
    CGContextRef context = UIGraphicsGetCurrentContext();
    
    CGContextSetFillColorWithColor(context, [color CGColor]);
    CGContextFillRect(context, rect);
    
    UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: MediaPlayer/MediaPlayer/Util/Category/UIButton+BackgroundColor.swift
//
//  UIButton+BackgroundColor.swift
//  USchool
//
//  Created by 龚昕炜 on 2017/9/15.
//  Copyright © 2017年 topglobaledu. All rights reserved.
//

import UIKit

extension UIButton {
    func setBackgroundColor(backgroundColor: UIColor, state: UIControlState) {
        self.setBackgroundImage(UIButton.p_imageWithColor(color: backgroundColor), for: state)
    }
    
    private class func p_imageWithColor(color: UIColor) -> UIImage {
        let rect = CGRect.init(x: 0.0, y: 0.0, width: 1.0, height: 1.0)
        UIGraphicsBeginImageContext(rect.size)
        let context = UIGraphicsGetCurrentContext()
        context?.setFillColor(color.cgColor)
        context?.fill(rect)
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return image!
    }
}
